commit 5b7a7b9eab981c08421c6c64f0a263e510025c3a (refs/changes/06/2206/2)
Author: Alain O'Dea <alain.odea@gmail.com>
Date:   2017-07-11T15:23:21-02:30 (2 years, 3 months ago)
    
    :memo: Document adding FDUM to SmartOS USB key
    
     - Add warning to makefile that it needs SmartOS to build
    
     - Add example of adding FDUM to SmartOS USB key to README

diff --git a/GNUmakefile b/GNUmakefile
index c9b58d3..3b93e9f 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -8,6 +8,8 @@
 # Copyright (c) 2014, Joyent, Inc.
 #
 
+# WARNING: this has only been tested to build on SmartOS
+
 NAME =	sdcboot
 
 TOP :=	$(shell pwd)
diff --git a/README.md b/README.md
index ec6f6e2..8c8010b 100644
--- a/README.md
+++ b/README.md
@@ -17,7 +17,50 @@ contribution guidelines, issues, and general documentation, visit the main
 Note that most of the contents of this repository are covered by their
 own separate licenses and copyright statements.
 
-# Development
+# Description
+
+Bootable **Firmware Diagnostics and Upgrade Mode** to facilitate manual
+BIOS or firmware upgrades or diagnostics that require a DOS-compatible
+environment. This is useful for, but independent of Triton and SmartOS.
+
+# Example: Add FreeDOS FDUM to SmartOS USB Key
+
+## Pre-requisites
+
+- multiarch SmartOS base zone
+- build-essential from PKGSRC
+
+## Procedure
+
+ 1. Run **make** on a multiarch SmartOS zone.
+ 2. Copy **proto/boot/memdisk** and **proto/boot/freedos.img** to
+    **/boot/grub** on the USB key.
+ 3. Copy **proto/dos** to **/dos** on the USB key.
+ 4. Add the following to **/boot/grub/menu.lst**:
+
+    ```
+    title Firmware Diagnostics and Update Mode [FreeDOS]
+       kernel /boot/grub/memdisk
+       module /boot/grub/freedos.img
+    ```
+
+ 5. Create a **firmware** directory on the root of the USB key.
+ 5. Copy DOS-compatible **BIOS.exe** to **/firmware** on the USB key.
+
+## Usage
+
+ 1. Boot computer from USB
+ 1. Select **Firmware Diagnostics and Update Mode [FreeDOS]**.
+
+    If **firmware/main.bat** is on the USB key it will run that
+    automatically. If you want to reboot after your firmware update, end
+    that script with `fdapm warmboot`. Otherwise, it will run your
+    script again in a loop.
+
+ 2. Run your BIOS Update **C:\firmware\BIOS.exe** (or whatever you named
+    it).
+
+# Contents
 
 This repository contains the following:
 
@@ -29,6 +72,8 @@ This repository contains the following:
 - A replacement for the BIOS int 0x14 service that actually works
 - Glue for building the FreeDOS-related portions of the SDC boot system
 
+# Purpose
+
 The repository serves three main purposes: first, it contains the
 original sources for all DOS-related binaries.  Since we don't have any
 ability to build them, they aren't really useful, but we still need them
